draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -848,6 +851,151 @@ Content-Type: text/plain
 </artwork>
 </section>
 
+<section title="Origin" anchor="http.origin">
+<t>
+   The origin for a given URI is the triple of scheme, host, and port
+   after normalizing the scheme and host to lowercase and normalizing the
+   port to remove any leading zeros. If port is elided from the URI, the
+   default port for that scheme is used. For example, the URI
+</t>
+<artwork type="example">
+   https://Example.Com/happy.js
+</artwork>
+<t>
+   would have the origin
+</t>
+<artwork type="example">
+   { "https", "example.com", "443" }
+</artwork>
+<t>
+   which can also be described as the normalized URI prefix with port always
+   present:
+</t>
+<artwork type="example">
+   https://example.com:443
+</artwork>
+<t>
+   The notion of an origin for any given URI is central to the semantics of
+   HTTP. Each origin defines its own namespace and controls how identifiers
+   within that namespace are mapped to resources. In turn, how the origin
+   responds to valid requests, consistently over time, determines the
+   semantics that users will associate with a URI, and the usefulness of
+   those semantics is what ultimately transforms these mechanisms into a
+   "resource" for users to reference and access in the future.
+</t>
+<t>
+   Two origins are distinct if they differ in scheme, host, or port. Even
+   when it can be verified that the same entity controls two distinct origins,
+   the two namespaces under those origins are distinct unless explicitly
+   aliased by a server authoritative for that origin.
+</t>
+<t>
+   Origin is also used within HTML and related Web protocols, beyond the
+   scope of this document, as described in <xref target="RFC6454"/>.
+</t>
+<t>
+   Note that the presence of a URI does not imply that there is always an
+   HTTP server at the identified origin listening for connections on
+   the identified host and port. Anyone can mint a URI. What the origin
+   determines is who has the right to respond authoritatively to requests that
+   target the identified resource. The delegated nature of registered names
+   and IP addresses creates a federated namespace whether or not an HTTP
+   server is present.
+</t>
+<t>
```

Comment:
It seems to me that this might be a good point to have a section that is dedicated to discussion of "http" origins.

Reply:
yes, it was running long. I am also going to look at moving the whole thing to Routing.